texthero/representation.py

```diff
@@ -542,11 +438,18 @@ def nmf(s, n_components=2, random_state=None) -> pd.Series:
 
     """
     nmf = NMF(n_components=n_components, init="random", random_state=random_state,)
-    return pd.Series(list(nmf.fit_transform(list(s))), index=s.index)
+
+    if _check_is_valid_DocumentTermDF(s):
+        s_coo = s.sparse.to_coo()
+        s_for_vectorization = s_coo.astype("float64")
+    else:
+        s_for_vectorization = list(s)
+
+    return pd.Series(list(nmf.fit_transform(s_for_vectorization)), index=s.index)
 
 
 def tsne(
-    s: pd.Series,
+    s: Union[pd.Series, pd.DataFrame],
```

Comment:
Calling `s` a variable that accepts both `s`eries and Dataframe is confusing ... ideas for a better name?

Reply:
I totally agree with you there. I changed the name to `input_matrix`, as we always get a matrix as input but in different representations, so instead of calling it similar to the datatype we called it `input_matrix` as we get a matrix input. 🇫🇯